Firestore/Source/Model/FSTDocumentKey.mm

```diff
@@ -21,26 +21,30 @@
 
 #import "Firestore/Source/Core/FSTFirestoreClient.h"
 
-#include "Firestore/core/src/firebase/firestore/model/resource_path.h"
+#include "Firestore/core/src/firebase/firestore/model/document_key.h"
 #include "Firestore/core/src/firebase/firestore/util/hard_assert.h"
 #include "Firestore/core/src/firebase/firestore/util/hashing.h"
 #include "Firestore/core/src/firebase/firestore/util/string_apple.h"
 
 namespace util = firebase::firestore::util;
 using firebase::firestore::model::ResourcePath;
+using firebase::firestore::model::DocumentKey;
 
 NS_ASSUME_NONNULL_BEGIN
 
 @interface FSTDocumentKey () {
-  /** The path to the document. */
-  ResourcePath _path;
+  DocumentKey _impl;
```

Comment:
This may be better named `_delegate`. I'm not a fan of "impl" as a name since it's generic without suggesting how it's the implementation.

Reply:
Done.